kratos/utilities/read_materials_utility.cpp

```diff
@@ -81,76 +88,113 @@ ReadMaterialsUtility::ReadMaterialsUtility(
+{
+    KRATOS_TRY;
+
     std::stringstream ss(rLine);
     std::size_t counter = 0;
     while (std::getline(ss, rLine, '.')){++counter;}
     KRATOS_WARNING_IF("Read materials", counter > 1) << "Ignoring module information for component " << rLine << std::endl;
+
+    KRATOS_CATCH("");
 }
 
 /***********************************************************************************/
 /***********************************************************************************/
 
-void ReadMaterialsUtility::AssignPropertyBlock(Parameters Data)
+void ReadMaterialsUtility::CreateProperty(
+    Parameters Data,
+    Properties::Pointer& pNewProperty,
+    const bool TreatAsSubproperty
+    )
 {
+    KRATOS_TRY;
+
     // Get the properties for the specified model part.
-    ModelPart& r_model_part = mrModel.GetModelPart(Data["model_part_name"].GetString());
-    const IndexType property_id = Data["properties_id"].GetInt();
-    const IndexType mesh_id = 0;
     Properties::Pointer p_prop;
-    if (r_model_part.RecursivelyHasProperties(property_id, mesh_id)) {
-        p_prop = r_model_part.pGetProperties(property_id, mesh_id);
-    } else {
-        p_prop = r_model_part.CreateNewProperties(property_id, mesh_id);
-    }
+    if (!TreatAsSubproperty) {
+        ModelPart& r_model_part = mrModel.GetModelPart(Data["model_part_name"].GetString());
+        const IndexType property_id = Data["properties_id"].GetInt();
+        const IndexType mesh_id = 0;
+        if (r_model_part.RecursivelyHasProperties(property_id, mesh_id)) {
+            p_prop = r_model_part.pGetProperties(property_id, mesh_id);
+        } else {
+            p_prop = r_model_part.CreateNewProperties(property_id, mesh_id);
+        }
 
-    // Assign the p_properties to the model part's elements and conditions.
-    auto& r_elements_array = r_model_part.Elements();
-    auto& r_conditions_array = r_model_part.Conditions();
+        // Assign the p_properties to the model part's elements and conditions.
+        auto& r_elements_array = r_model_part.Elements();
+        auto& r_conditions_array = r_model_part.Conditions();
 
-    #pragma omp parallel for
-    for(int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
-        auto it_elem = r_elements_array.begin() + i;
-        it_elem->SetProperties(p_prop);
-    }
+        #pragma omp parallel for
+        for(int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
+            auto it_elem = r_elements_array.begin() + i;
+            it_elem->SetProperties(p_prop);
+        }
 
-    #pragma omp parallel for
-    for(int i = 0; i < static_cast<int>(r_conditions_array.size()); ++i) {
-        auto it_cond = r_conditions_array.begin() + i;
-        it_cond->SetProperties(p_prop);
+        #pragma omp parallel for
+        for(int i = 0; i < static_cast<int>(r_conditions_array.size()); ++i) {
+            auto it_cond = r_conditions_array.begin() + i;
+            it_cond->SetProperties(p_prop);
```

Comment:
assigning the property was already done before in `AssignPropertyBlock`, if you have to repeat it here then I believe sth is wrong

Reply:
Maybe it is a leftover of some conflict